### PR TITLE
battery bound

### PR DESCRIPTION
**Pull request by gazzayeah**

### CODE REVIEW

**Comment by gazzayeah** [comment]
Set battery charging rate between pilot signal and 0.

File: acnportal/acnsim/models/battery.py [hunk 1]
```diff
@@ -156,13 +156,13 @@ def charge(self, pilot, voltage, period):
...
             if self._noise_level > 0:
-                charge_power -= abs(np.random.normal(0, self._noise_level))
+                charge_power = max(charge_power - abs(np.random.normal(0, self._noise_level)), 0)
         else:
             charge_power = min([pilot * voltage / 1000,
                                 (1 - self._soc) / (1 - self._transition_soc) * self._max_power,
                                 rate_to_full])
             if self._noise_level > 0:
-                charge_power += np.random.normal(0, self._noise_level)
+                charge_power = min(max(charge_power + np.random.normal(0, self._noise_level), 0), pilot * voltage / 1000, self._max_power, rate_to_full)
```
Comment (zach401) [inline acnportal/acnsim/models/battery.py:165]: For clarify, perhaps we should use numpy.clip (https://numpy.org/doc/stable/reference/generated/numpy.clip.html?highlight=clip#numpy.clip)

For example we could say:
```suggestion
                upper_bound = min(pilot * voltage / 1000, self._max_power, rate_to_full)
                charge_power += np.random.normal(0, self._noise_level)
                charge_power = np.clip(charge_power, 0, upper_bound)
```

Comment (zach401) [reply acnportal/acnsim/models/battery.py:165]: We should also probably add some tests in tests/test_battery.py, to make sure this works as we expect.

File: acnportal/acnsim/models/battery.py [hunk 2]
```diff
@@ -156,13 +156,13 @@ def charge(self, pilot, voltage, period):
...
                 # ensure that noise does not cause the battery to violate any hard limits.
                 charge_power = min([charge_power, pilot * voltage / 1000, self._max_power, rate_to_full])
```
Comment (zach401) [inline acnportal/acnsim/models/battery.py:167]: With the new addition, this line is redundant.

**Comment by gazzayeah** [comment]
Hi Zachary,

I apologize for my carelessness. The last line should indeed be removed and was left in the code unintentionally. I’ll make the code more concise and test it before commit.

Lots of thanks,
Gary Ye

**Comment by zach401** [comment]
Hi Gary, 

Don't worry about it! Let me know if you need any help with writing tests. Thanks again for finding this bug and fixing it. Glad to have you contributing to the project.

**Pull request merged**